addons/web/static/src/js/views/calendar/calendar_view.js

```diff
@@ -151,17 +151,28 @@ var CalendarView = AbstractView.extend({
             }
         }
 
+        let allowedScales;
+        const possibleScales = ['day', 'week', 'month', 'year'];
+        if (arch.attrs.scales) {
+            allowedScales = arch.attrs.scales.split(',')
+                .filter(x => possibleScales.includes(x));
+        } else {
+            allowedScales = possibleScales;
+        }
+
         this.controllerParams.eventOpenPopup = utils.toBoolElse(attrs.event_open_popup || '', false);
         this.controllerParams.showUnusualDays = utils.toBoolElse(attrs.show_unusual_days || '', false);
         this.controllerParams.mapping = mapping;
         this.controllerParams.context = params.context || {};
         this.controllerParams.displayName = params.action && params.action.name;
+        this.controllerParams.allowedScales = allowedScales;
 
         this.rendererParams.displayFields = displayFields;
         this.rendererParams.model = viewInfo.model;
         this.rendererParams.hideDate = utils.toBoolElse(attrs.hide_date || '', false);
         this.rendererParams.hideTime = utils.toBoolElse(attrs.hide_time || '', false);
         this.rendererParams.canDelete = this.controllerParams.activeActions.delete;
+        this.rendererParams.canCreate = this.controllerParams.activeActions.create;
```

Comment:
arf
not linked to this commit, linked to "[IMP] base,web: add year scale to calendar"